CI: tests/drivers/bbram/drivers.bbram.stm32_rtc fails to build on stm32f0_disco due to bbram_stm32.c compilation error

**Describe the bug**
`tests/drivers/bbram/drivers.bbram.stm32_rtc` fails to build on `stm32f0_disco` due to `bbram_stm32.c` compilation error as in https://github.com/zephyrproject-rtos/zephyr/actions/runs/7859262283/job/21474045907

**To Reproduce**
Steps to reproduce the behavior:
1. `west build -p -b stm32f0_disco tests/drivers/bbram -T drivers.bbram.stm32_rtc` 
2. See error

**Expected behavior**
CI build succeeds.

**Impact**
CI build failure.

**Logs and console output**
```
$ west build -p -b stm32f0_disco tests/drivers/bbram -T drivers.bbram.stm32_rtc
-- west build: making build dir /home/brix/Projects/zephyrproject/zephyr/build pristine
-- west build: generating a build system
Loading Zephyr default modules (Zephyr base).
-- Application: /home/brix/Projects/zephyrproject/zephyr/tests/drivers/bbram
-- CMake version: 3.28.1
-- Found Python3: /home/brix/Projects/zephyrproject/venv/bin/python3 (found suitable version "3.10.12", minimum required is "3.8") found components: Interpreter 
-- Cache files will be written to: /home/brix/.cache/zephyr
-- Zephyr version: 3.6.0-rc2 (/home/brix/Projects/zephyrproject/zephyr)
-- Found west (found suitable version "1.2.0", minimum required is "0.14.0")
-- Board: stm32f0_disco
-- ZEPHYR_TOOLCHAIN_VARIANT not set, trying to locate Zephyr SDK
-- Found host-tools: zephyr 0.16.5 (/opt/zephyr-sdk/0.16.5)
-- Found toolchain: zephyr 0.16.5 (/opt/zephyr-sdk/0.16.5)
-- Found Dtc: /opt/zephyr-sdk/0.16.5/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6") 
-- Found BOARD.dts: /home/brix/Projects/zephyrproject/zephyr/boards/arm/stm32f0_disco/stm32f0_disco.dts
-- Generated zephyr.dts: /home/brix/Projects/zephyrproject/zephyr/build/zephyr/zephyr.dts
-- Generated devicetree_generated.h: /home/brix/Projects/zephyrproject/zephyr/build/zephyr/include/generated/devicetree_generated.h
-- Including generated dts.cmake file: /home/brix/Projects/zephyrproject/zephyr/build/zephyr/dts.cmake
Parsing /home/brix/Projects/zephyrproject/zephyr/Kconfig
Loaded configuration '/home/brix/Projects/zephyrproject/zephyr/boards/arm/stm32f0_disco/stm32f0_disco_defconfig'
Merged configuration '/home/brix/Projects/zephyrproject/zephyr/tests/drivers/bbram/prj.conf'
Merged configuration '/home/brix/Projects/zephyrproject/zephyr/tests/drivers/bbram/stm32_rtc.conf'
Configuration saved to '/home/brix/Projects/zephyrproject/zephyr/build/zephyr/.config'
Kconfig header saved to '/home/brix/Projects/zephyrproject/zephyr/build/zephyr/include/generated/autoconf.h'
-- Found GnuLd: /opt/zephyr-sdk/0.16.5/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd (found version "2.38") 
-- The C compiler identification is GNU 12.2.0
-- The CXX compiler identification is GNU 12.2.0
-- The ASM compiler identification is GNU
-- Found assembler: /opt/zephyr-sdk/0.16.5/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
-- Using ccache: /usr/bin/ccache
-- Configuring done (3.6s)
-- Generating done (0.1s)
-- Build files have been written to: /home/brix/Projects/zephyrproject/zephyr/build
-- west build: building application
[  1%] Generating misc/generated/syscalls.json, misc/generated/struct_tags.json
[  2%] Built target parse_syscalls_target
[  3%] Generating include/generated/kobj-types-enum.h, include/generated/otype-to-str.h, include/generated/otype-to-size.h
[  3%] Built target kobj_types_h_target
[  4%] Generating include/generated/syscall_dispatch.c, include/generated/syscall_list.h
[  4%] Built target syscall_list_h_target
[  4%] Generating include/generated/driver-validation.h
[  4%] Built target driver_validation_h_target
[  5%] Building C object zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj
[  5%] Built target offsets
[  5%] Generating include/generated/offsets.h
[  5%] Built target offsets_h
[  5%] Generating include/generated/version.h
-- Zephyr version: 3.6.0-rc2 (/home/brix/Projects/zephyrproject/zephyr), build: v3.6.0-rc2-50-g61f065230ad9
[  5%] Built target version_h
[  5%] Built target zephyr_generated_headers
[  5%] Building C object CMakeFiles/app.dir/src/main.c.obj
[  6%] Linking C static library app/libapp.a
[  6%] Built target app
[  7%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/heap/heap.c.obj
[  8%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/cbprintf_packaged.c.obj
[  8%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/printk.c.obj
[  9%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/sem.c.obj
[ 10%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/thread_entry.c.obj
[ 11%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/cbprintf_complete.c.obj
[ 11%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/assert.c.obj
[ 12%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/reboot.c.obj
[ 13%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/utils/dec.c.obj
[ 13%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/utils/hex.c.obj
[ 14%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/utils/rb.c.obj
[ 15%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/utils/timeutil.c.obj
[ 15%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/utils/bitarray.c.obj
[ 16%] Building C object zephyr/CMakeFiles/zephyr.dir/misc/generated/configs.c.obj
[ 17%] Building C object zephyr/CMakeFiles/zephyr.dir/soc/arm/st_stm32/stm32f0/soc.c.obj
[ 18%] Building C object zephyr/CMakeFiles/zephyr.dir/soc/arm/st_stm32/common/stm32cube_hal.c.obj
[ 18%] Building C object zephyr/CMakeFiles/zephyr.dir/soc/arm/st_stm32/common/soc_config.c.obj
[ 19%] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/logging/log_minimal.c.obj
[ 20%] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/tracing/tracing_none.c.obj
[ 20%] Linking C static library libzephyr.a
[ 20%] Built target zephyr
[ 20%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/main_weak.c.obj
[ 21%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/banner.c.obj
[ 22%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/busy_wait.c.obj
[ 22%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/device.c.obj
[ 23%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/errno.c.obj
[ 25%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/fatal.c.obj
[ 26%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/init.c.obj
[ 26%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/kheap.c.obj
[ 27%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/mem_slab.c.obj
[ 28%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/thread.c.obj
[ 28%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/version.c.obj
[ 29%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/sched.c.obj
[ 30%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/idle.c.obj
[ 30%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/mailbox.c.obj
[ 31%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/msg_q.c.obj
[ 32%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/mutex.c.obj
[ 33%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/queue.c.obj
[ 33%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/sem.c.obj
[ 34%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/stack.c.obj
[ 35%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/system_work_q.c.obj
[ 35%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/work.c.obj
[ 36%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/condvar.c.obj
[ 37%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/xip.c.obj
[ 37%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/timeout.c.obj
[ 38%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/timer.c.obj
[ 39%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/atomic_c.c.obj
[ 40%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/mempool.c.obj
[ 40%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/dynamic_disabled.c.obj
[ 41%] Linking C static library libkernel.a
[ 41%] Built target kernel
[ 42%] Generating linker_zephyr_pre0.cmd
[ 42%] Built target linker_zephyr_prebuilt_script_target
[ 43%] Building C object zephyr/arch/common/CMakeFiles/isr_tables.dir/isr_tables.c.obj
[ 44%] Linking C static library libisr_tables.a
[ 44%] Built target isr_tables
[ 45%] Building C object zephyr/arch/common/CMakeFiles/arch__common.dir/sw_isr_common.c.obj
[ 46%] Linking C static library libarch__common.a
[ 46%] Built target arch__common
[ 47%] Building C object zephyr/arch/arch/arm/core/CMakeFiles/arch__arm__core.dir/fatal.c.obj
[ 48%] Building C object zephyr/arch/arch/arm/core/CMakeFiles/arch__arm__core.dir/nmi.c.obj
[ 48%] Building ASM object zephyr/arch/arch/arm/core/CMakeFiles/arch__arm__core.dir/nmi_on_reset.S.obj
[ 50%] Building C object zephyr/arch/arch/arm/core/CMakeFiles/arch__arm__core.dir/tls.c.obj
[ 51%] Linking C static library libarch__arm__core.a
[ 51%] Built target arch__arm__core
[ 51%] Building ASM object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/exc_exit.S.obj
[ 52%] Building C object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/fault.c.obj
[ 53%] Building ASM object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/fault_s.S.obj
[ 53%] Building C object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/fpu.c.obj
[ 54%] Building ASM object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/reset.S.obj
[ 55%] Building C object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/scb.c.obj
[ 56%] Building C object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/thread_abort.c.obj
[ 56%] Building ASM object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/vector_table.S.obj
[ 57%] Building C object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/swap.c.obj
[ 58%] Building ASM object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/swap_helper.S.obj
[ 58%] Building C object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/irq_manage.c.obj
[ 59%] Building C object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/prep_c.c.obj
[ 60%] Building C object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/thread.c.obj
[ 60%] Building ASM object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/cpu_idle.S.obj
[ 61%] Building C object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/irq_init.c.obj
[ 62%] Building ASM object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/isr_wrapper.S.obj
[ 63%] Building ASM object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/__aeabi_read_tp.S.obj
[ 63%] Linking C static library libarch__arm__core__cortex_m.a
[ 63%] Built target arch__arm__core__cortex_m
[ 63%] Building C object zephyr/lib/libc/picolibc/CMakeFiles/lib__libc__picolibc.dir/libc-hooks.c.obj
[ 64%] Linking C static library liblib__libc__picolibc.a
[ 64%] Built target lib__libc__picolibc
[ 64%] Building C object zephyr/lib/libc/common/CMakeFiles/lib__libc__common.dir/source/stdlib/abort.c.obj
[ 65%] Building C object zephyr/lib/libc/common/CMakeFiles/lib__libc__common.dir/source/stdlib/malloc.c.obj
[ 66%] Linking C static library liblib__libc__common.a
[ 66%] Built target lib__libc__common
[ 67%] Building C object zephyr/subsys/testsuite/ztest/CMakeFiles/subsys__testsuite__ztest.dir/src/ztest.c.obj
[ 68%] Building C object zephyr/subsys/testsuite/ztest/CMakeFiles/subsys__testsuite__ztest.dir/src/ztest_error_hook.c.obj
[ 69%] Building C object zephyr/subsys/testsuite/ztest/CMakeFiles/subsys__testsuite__ztest.dir/src/ztest_rules.c.obj
[ 69%] Building C object zephyr/subsys/testsuite/ztest/CMakeFiles/subsys__testsuite__ztest.dir/src/ztest_defaults.c.obj
[ 70%] Linking C static library libsubsys__testsuite__ztest.a
[ 70%] Built target subsys__testsuite__ztest
[ 70%] Building C object zephyr/drivers/interrupt_controller/CMakeFiles/drivers__interrupt_controller.dir/intc_exti_stm32.c.obj
[ 71%] Linking C static library libdrivers__interrupt_controller.a
[ 71%] Built target drivers__interrupt_controller
[ 72%] Building C object zephyr/drivers/bbram/CMakeFiles/drivers__bbram.dir/bbram_stm32.c.obj
In file included from /home/brix/Projects/zephyrproject/zephyr/include/zephyr/sys/util_macro.h:34,
                 from /home/brix/Projects/zephyrproject/zephyr/include/zephyr/irq_multilevel.h:15,
                 from /home/brix/Projects/zephyrproject/zephyr/include/zephyr/devicetree.h:20,
                 from /home/brix/Projects/zephyrproject/zephyr/include/zephyr/device.h:12,
                 from /home/brix/Projects/zephyrproject/zephyr/include/zephyr/drivers/bbram.h:12,
                 from /home/brix/Projects/zephyrproject/zephyr/drivers/bbram/bbram_stm32.c:11:
/home/brix/Projects/zephyrproject/zephyr/drivers/bbram/bbram_stm32.c:23:51: error: unknown type name 'RTC_TypeDef'
   23 | #define STM32_BKP_REG_OFFSET             offsetof(RTC_TypeDef, BKP0R)
      |                                                   ^~~~~~~~~~~
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:72:26: note: in definition of macro '__DEBRACKET'
   72 | #define __DEBRACKET(...) __VA_ARGS__
      |                          ^~~~~~~~~~~
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET'
   64 |         __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code)
      |         ^~~~~~~~~~~~~~~~~~~~
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE'
   59 |         __COND_CODE(_XXXX##_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/sys/util_macro.h:180:9: note: in expansion of macro 'Z_COND_CODE_1'
  180 |         Z_COND_CODE_1(_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~~~
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/devicetree.h:4260:9: note: in expansion of macro 'COND_CODE_1'
 4260 |         COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT),   \
      |         ^~~~~~~~~~~
/home/brix/Projects/zephyrproject/zephyr/drivers/bbram/bbram_stm32.c:108:66: note: in expansion of macro 'STM32_BKP_REG_OFFSET'
  108 |                 .base_addr = DT_REG_ADDR(DT_INST_PARENT(inst)) + STM32_BKP_REG_OFFSET,             \
      |                                                                  ^~~~~~~~~~~~~~~~~~~~
/home/brix/Projects/zephyrproject/zephyr/build/zephyr/include/generated/devicetree_generated.h:9395:49: note: in expansion of macro 'BBRAM_INIT'
 9395 | #define DT_FOREACH_OKAY_INST_st_stm32_bbram(fn) fn(0)
      |                                                 ^~
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_FOREACH_OKAY_INST_st_stm32_bbram'
  105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__
      |                                    ^
/home/brix/Projects/zephyrproject/zephyr/drivers/bbram/bbram_stm32.c:114:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY'
  114 | DT_INST_FOREACH_STATUS_OKAY(BBRAM_INIT);
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
gmake[2]: *** [zephyr/drivers/bbram/CMakeFiles/drivers__bbram.dir/build.make:76: zephyr/drivers/bbram/CMakeFiles/drivers__bbram.dir/bbram_stm32.c.obj] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:3616: zephyr/drivers/bbram/CMakeFiles/drivers__bbram.dir/all] Error 2
gmake: *** [Makefile:91: all] Error 2
FATAL ERROR: command exited with status 2: /home/brix/Projects/zephyrproject/venv/bin/cmake --build /home/brix/Projects/zephyrproject/zephyr/build
```

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain Zephyr SDK
 - Commit SHA: 61f065230ad9bca91633cd57b9dcd5e90c5d8bc7

--- a/drivers/bbram/bbram_stm32.c
+++ b/drivers/bbram/bbram_stm32.c
@@ -11,6 +11,7 @@
 #include <zephyr/drivers/bbram.h>
 #include <zephyr/logging/log.h>
 #include <zephyr/sys/util.h>
+#include <stm32_ll_rtc.h>
 LOG_MODULE_REGISTER(bbram, CONFIG_BBRAM_LOG_LEVEL);
 
 #define STM32_BKP_REG_BYTES		 4

